workflow/validate/validate.go

```diff
@@ -198,19 +204,17 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	return wfConditions, nil
 }
 
-// ValidateWorkflow accepts a workflow template and performs validation against it.
-func ValidateWorkflowTemplate(wftmplGetter templateresolution.WorkflowTemplateNamespacedGetter, cwftmplGetter templateresolution.ClusterWorkflowTemplateGetter, wftmpl wfv1.TemplateGetter) error {
-	ctx := newTemplateValidationCtx(nil, ValidateOpts{})
-	tmplCtx := templateresolution.NewContext(wftmplGetter, cwftmplGetter, wftmpl, nil)
+// ValidateWorkflowTemplate accepts a workflow template and performs validation against it.
+func ValidateWorkflowTemplate(wftmplGetter templateresolution.WorkflowTemplateNamespacedGetter, cwftmplGetter templateresolution.ClusterWorkflowTemplateGetter, wftmpl *wfv1.WorkflowTemplate) (*wfv1.WorkflowConditions, error) {
+	wf := common.ConvertWorkflowTemplateToWorkflow(wftmpl)
+	return ValidateWorkflow(wftmplGetter, cwftmplGetter, wf, ValidateOpts{IgnoreEntryPoint: true})
+}
+
+// ValidateClusterWorkflowTemplate accepts a cluster workflow template and performs validation against it.
+func ValidateClusterWorkflowTemplate(wftmplGetter templateresolution.WorkflowTemplateNamespacedGetter, cwftmplGetter templateresolution.ClusterWorkflowTemplateGetter, cwftmpl *wfv1.ClusterWorkflowTemplate) (*wfv1.WorkflowConditions, error) {
+	wf := common.ConvertClusterWorkflowTemplateToWorkflow(cwftmpl)
+	return ValidateWorkflow(wftmplGetter, cwftmplGetter, wf, ValidateOpts{IgnoreEntryPoint: true})
 
-	// Check if all templates can be resolved.
-	for _, template := range wftmpl.GetAllTemplates() {
```

Comment:
Two things here:

1. This function was changed to `wftmpl.GetTemplates()`in #2573. The fact that it hasn't changed here indicates that you haven't merged with master in a while. Can you do so?

2. Since this is the only place where `GetTemplates()` is used, and now it seems like it's not longer needed, can yo also remove it from the `TemplateHolder` interface and the `Workflow`, `WorkflowTemplate`, and `ClusterWorkflowTemplate` types?